tests/playwright/src/model/pages/resource-card-page.ts

```diff
@@ -0,0 +1,40 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import { type Locator, type Page } from '@playwright/test';
+
+import { ResourcesPage } from './resources-page';
+
+export class ResourceCardPage extends ResourcesPage {
+  readonly resourceCard: Locator;
+  readonly providerSetup: Locator;
+  readonly providerConnections: Locator;
+  readonly markdownContent: Locator;
+
+  constructor(page: Page, resourceName: string) {
+    super(page);
+    this.resourceCard = this.featuredProviderResources.getByRole('region', { name: resourceName, exact: true });
+    this.providerSetup = this.resourceCard.getByRole('region', { name: 'Provider Setup', exact: true });
+    this.providerConnections = this.resourceCard.getByRole('region', { name: 'Provider Connections', exact: true });
+    this.markdownContent = this.providerConnections.getByLabel('markdown-content');
+  }
+
+  public async resourceCardIsVisible(): Promise<boolean> {
```

Comment:
This method should be part of `ResourcesPage`, when writing a test, we will go from the top level to the bottom level.